modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/DoublesScheduleGenerator.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.credit.cds;
+
+import java.util.Arrays;
+
+import com.opengamma.strata.collect.array.DoubleArray;
+
+/**
+ * The Doubles schedule generator.
+ * <p>
+ * This is a utility for combining and truncating ascending arrays of doubles.
+ * The main application of this class is {@code IsdaCdsProductPricer}.
+ */
+public class DoublesScheduleGenerator {
+  private static final double TOL = 1d / 730d;
+
+  /**
+   * Combines the discount curve nodes and credit curve nodes. 
+   * <p>
+   * The combined numbers are returned only if the values strictly between the specified start and end. 
+   * <p>
+   * The start and end values are added at the beginning and end of the list. 
+   * If two times are very close (defined as less than half a day - 1/730 years different) only the smaller value is kept 
+   * (with the exception of the end value which takes precedence). 
+   * 
+   * @param start  the first time in the list
+   * @param end  the last time in the list
+   * @param discountCurveNodes  the discount curve node set
+   * @param creditCurveNodes  the credit curve node
+   * @return the combined list between start and end
+   */
+  static DoubleArray getIntegrationsPoints(
+      double start,
+      double end,
+      DoubleArray discountCurveNodes,
+      DoubleArray creditCurveNodes) {
+
+    double[] set1 = truncateSetExclusive(start, end, discountCurveNodes.toArray());
+    double[] set2 = truncateSetExclusive(start, end, creditCurveNodes.toArray());
+    int n1 = set1.length;
+    int n2 = set2.length;
+    int n = n1 + n2;
+    double[] set = new double[n];
+    System.arraycopy(set1, 0, set, 0, n1);
+    System.arraycopy(set2, 0, set, n1, n2);
+    Arrays.sort(set);
+
+    double[] temp = new double[n + 2];
+    temp[0] = start;
+    int pos = 0;
+    for (int i = 0; i < n; i++) {
+      if (different(temp[pos], set[i])) {
+        temp[++pos] = set[i];
+      }
+    }
+    if (different(temp[pos], end)) {
+      pos++;
+    }
+    temp[pos] = end; // add the end point (this may replace the last entry in temp if that is not significantly different)
+    int resLength = pos + 1;
+    return DoubleArray.copyOf(temp, 0, resLength);
+  }
+
+  private static boolean different(double a, double b) {
+    return Math.abs(a - b) > TOL;
+  }
+
+  private static double[] truncateSetExclusive(double lower, double upper, double[] set) {
```

Comment:
The result should at a minimum say that it can be modified (it is by `truncateSetInclusive`). Without this, there would be the opportunity to use `toArrayUnsafe()` on the inputs.